photutils/psf/photometry.py

```diff
@@ -274,10 +278,17 @@ def do_photometry(self, image, positions=None):
                                   sources['ycentroid'],
                                   sources['aperture_flux']])
 
+        self._pars_to_set, self._pars_to_output = self._define_fit_param_names()
```

Comment:
Thinking about this a bit more.  It occurred to me that 3 has a failure mode that if the user changes which parameters are fixed, it renders these invalid... but there's no way for the photometry class to *know* that, because the model doesn't send any signal to the photometry class.

So that favors 1 or 2 over 3...